diff -urN libfreebob-1.0.11/src/libfreebobavc/avc_extended_plug_info.cpp libfreebob-1.0.11.new/src/libfreebobavc/avc_extended_plug_info.cpp
--- libfreebob-1.0.11/src/libfreebobavc/avc_extended_plug_info.cpp	2007-10-15 20:15:08.000000000 +0200
+++ libfreebob-1.0.11.new/src/libfreebobavc/avc_extended_plug_info.cpp	2008-07-12 23:20:35.001927537 +0200
@@ -23,6 +23,7 @@
 #include "ieee1394service.h"
 
 #include <cstdlib>
+#include <cstring>
 #include <netinet/in.h>
 #include <iostream>
 
diff -urN libfreebob-1.0.11/src/libfreebobavc/avc_generic.cpp libfreebob-1.0.11.new/src/libfreebobavc/avc_generic.cpp
--- libfreebob-1.0.11/src/libfreebobavc/avc_generic.cpp	2007-10-15 20:15:08.000000000 +0200
+++ libfreebob-1.0.11.new/src/libfreebobavc/avc_generic.cpp	2008-07-12 23:20:47.838592298 +0200
@@ -23,6 +23,7 @@
 #include "ieee1394service.h"
 
 #include <cstdlib>
+#include <cstring>
 #include <netinet/in.h>
 
 #define DEBUG_EXTRA_VERBOSE 2
diff -urN libfreebob-1.0.11/tests/test-extplugcmd.cpp libfreebob-1.0.11.new/tests/test-extplugcmd.cpp
--- libfreebob-1.0.11/tests/test-extplugcmd.cpp	2007-10-15 20:15:07.000000000 +0200
+++ libfreebob-1.0.11.new/tests/test-extplugcmd.cpp	2008-07-12 23:24:47.762337527 +0200
@@ -24,6 +24,7 @@
 #include "libfreebobavc/ieee1394service.h"
 
 #include <argp.h>
+#include <cstdlib>
 
 using namespace std;
 
diff -urN libfreebob-1.0.11/tests/test-volume.cpp libfreebob-1.0.11.new/tests/test-volume.cpp
--- libfreebob-1.0.11/tests/test-volume.cpp	2007-10-15 20:15:07.000000000 +0200
+++ libfreebob-1.0.11.new/tests/test-volume.cpp	2008-07-12 23:24:53.892526352 +0200
@@ -22,6 +22,8 @@
 #include "libfreebobavc/serialize.h"
 #include "libfreebobavc/ieee1394service.h"
 
+#include <cstdlib>
+
 const bool bVerbose = false;
 
 short int
